tests/agents/gherkin-specs/azure_functions_metadata.feature

```diff
@@ -0,0 +1,70 @@
+Feature: Extracting Metadata for Azure Function Apps
+
+  Background:
+    Given an agent configured with
+      | setting        | value |
+      | cloud_provider | azure |
+
+  Scenario Outline: Azure Function App with minimum set of environment variables present in expected format
+    Given the following environment variables are present
+      | name                        | value                                                                   |
+      | FUNCTIONS_EXTENSION_VERSION | version                                                                 |
+      | WEBSITE_OWNER_NAME          | d2cd53b3-acdc-4964-9563-3f5201556a81+faas_group-CentralUSwebspace-Linux |
+      | WEBSITE_SITE_NAME           | site_name                                                               |

```

Comment:
Before the gherkin spec was added (when I reviewed this PR), I took no notice of the `(fallback: ...` comments for `project.name` and `region`:

> | `project.name`        | `WEBSITE_RESOURCE_GROUP` (fallback: `{resource group}` from `WEBSITE_OWNER_NAME`) |
> | `region`              | `REGION_NAME`  (fallback: `{region}` from `WEBSITE_OWNER_NAME`) |

Do you know of a real Azure Function App case where `REGION_NAME` and `WEBSITE_RESOURCE_GROUP` are *not* defined? https://learn.microsoft.com/en-us/azure/app-service/reference-app-settings defines both of those envvars.

If we don't know of such a case, what do you think about *dropping* mention of the fallback to `WEBSITE_OWNER_NAME` and *dropping* this Gherkin scenario?

Reply:
Tbh I haven't had a case lately where this fallback would have been necessary.
Probably with older versions of the Azure Functions runtime.
I'll definitely leave that fallback in the .NET agent's code, but if you prefer, we can remove the test case.